src/sharedHooks/useAuthenticatedQuery.js

```diff
@@ -1,29 +1,60 @@
 import { useQuery } from "@tanstack/react-query";
-import { getJWT } from "components/LoginSignUp/AuthenticationService.ts";
+import { getJWT, isLoggedIn } from "components/LoginSignUp/AuthenticationService.ts";
+import { useEffect, useState } from "react";
+import { logWithoutRemote } from "sharedHelpers/logger";
 import { reactQueryRetry } from "sharedHelpers/logging";
 
+const localLogger = logWithoutRemote.extend( "useAuthenticatedQuery" );
+
+const LOGGED_IN_UNKNOWN = null;
+
 // Should work like React Query's useQuery except it calls the queryFunction
 // with an object that includes the JWT
 const useAuthenticatedQuery = (
   queryKey,
   queryFunction,
   queryOptions = {}
-) => useQuery( {
-  queryKey: [...queryKey, queryOptions.allowAnonymousJWT],
-  queryFn: async ( ) => {
-    // Note, getJWT() takes care of fetching a new token if the existing
-    // one is expired. We *could* store the token in state with useState if
-    // fetching from RNSInfo becomes a performance issue
-    const apiToken = await getJWT( queryOptions.allowAnonymousJWT );
-    const options = {
-      api_token: apiToken
-    };
-    return queryFunction( options );
-  },
-  retry: ( failureCount, error ) => reactQueryRetry( failureCount, error, {
-    queryKey
-  } ),
-  ...queryOptions
-} );
+) => {
+  const [userLoggedIn, setUserLoggedIn] = useState( LOGGED_IN_UNKNOWN );
```

Comment:
We check for login state in so many places throughout iNatNext that I'm often wondering if we should be leveraging react navigation or persisting the login state in store somehow. I guess the main question here -- is there a reason to write a custom `useEffect` instead of using the `useCurrentUser` hook to check for login that we use everywhere else?

I don't *quite* know what solution we could implement with react navigation, but here's their documentation on authentication flows that makes me curious about how else we could do this: https://reactnavigation.org/docs/auth-flow/#what-we-need

Reply:
`useCurrentUser()` was my first attempt here, but it only returns null or a Realm user object. Here, we need to not execute the query if we don't know whether the user is logged in or not, and `useCurrentUser` doesn't tell us that. It conflates the unknown state with the logged out state. I could have altered `useCurrentUser` to have a yes|no|dunno response, but that seemed more invasive than this work merited. Altering `useAuthenticatedQuery` seemed unvasive enough.